src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs

```diff
@@ -510,11 +510,13 @@ public async Task<IActionResult> EditPassword(string id)
 
         var model = new EditPasswordViewModel { UsernameOrEmail = user.UserName };
 
+        ViewData["ReturnUrl"] = returnUrl;
```

Comment:
You need to add this in the post method as well. Otherwise you loose it when a validation error is shown.